homeassistant/components/linky/__init__.py

```diff
@@ -1 +1,58 @@
 """The linky component."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DEFAULT_TIMEOUT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+ACCOUNT_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [ACCOUNT_SCHEMA]))},
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up Linky sensor from legacy config file."""
+
+    conf = config[DOMAIN]
+    if conf is not None:
+        for linky_account_conf in conf:
+
+            if (
+                linky_account_conf is not None
+                and not hass.config_entries.async_entries(DOMAIN)
```

Comment:
There can be more than one config entry for this domain, right? We need to check for some specific thing in the config of the to be config entry and not start the flow for already set up entries.

Reply:
Yep, deleting this.

And `async_step_import` is already checking if the `_configuration_exists`